src/Governor/GovernorGuardian.sol

```diff
@@ -0,0 +1,172 @@
+
+        for (uint256 i = 0; i < vaults.length; ++i) {
+            address vault = vaults[i];
+
+            if (!canBeUnpaused(vault, guardianCalling)) continue;
+
+            // Restore the hook configuration.
+            IEVault(vault).setHookConfig(pauseDatas[vault].hookTarget, pauseDatas[vault].hookedOps);
+
+            emit Unpaused(vault);
+        }
+    }
+
+    /// @notice Changes pause status of the selected operations for the given vaults.
+    /// @param vaults The array of vault addresses to be unpaused.
+    /// @param newHookedOps The new hooked operations bitmap.
+    function changePauseStatus(address[] calldata vaults, uint32 newHookedOps)
+        external
+        nonReentrant
+        onlyRole(GUARDIAN_ROLE)
+    {
+        for (uint256 i = 0; i < vaults.length; ++i) {
+            address vault = vaults[i];
+
+            if (!canPauseStatusChange(vault)) continue;
+
+            // Change the hook configuration, i.e. by unpausing the selected operations.
+            IEVault(vault).setHookConfig(address(0), newHookedOps);
+
+            emit PauseStatusChanged(vault);
+        }
+    }
+
+    /// @notice Checks if the given vault can be paused.
+    /// @param vault The address of the vault to check.
+    /// @return bool True if the vault can be paused, false otherwise.
+    function canBePaused(address vault) public view returns (bool) {
+        // The vault can be paused if the pause cooldown has passed since the last pause.
+        return pauseDatas[vault].lastPauseTimestamp + PAUSE_COOLDOWN < block.timestamp
+            && IEVault(vault).governorAdmin() == address(this);
+    }
+
+    /// @notice Checks if the given vault can be unpaused.
+    /// @param vault The address of the vault to check.
+    /// @param guardianCalling Whether the guardian is calling the function.
+    /// @return bool True if the vault can be unpaused, false otherwise.
+    function canBeUnpaused(address vault, bool guardianCalling) public view returns (bool) {
+        uint256 lastPauseTimestamp = pauseDatas[vault].lastPauseTimestamp;
+
+        // The vault can be unpaused if the guardian is calling the function or if the pause duration has passed.
+        // We must ensure that the hook config was previously cached by checking that the lastPauseTimestamp is not 0.
+        return lastPauseTimestamp != 0 && (guardianCalling || lastPauseTimestamp + PAUSE_DURATION < block.timestamp)
+            && IEVault(vault).governorAdmin() == address(this);
+    }
+
+    /// @notice Checks if the given vault status can be changed.
+    /// @param vault The address of the vault to check.
+    /// @return bool True if the vault status can be changed, false otherwise.
+    function canPauseStatusChange(address vault) public view returns (bool) {
+        PauseData memory pauseData = pauseDatas[vault];
+
+        // The pause status can be changed if the cached hook target is the zero address to prevent vault
```

Comment:
I guess this means you wouldn't be able to selectively disable a vulnerable method on a vault that uses hooks.

What misconfigurations are you worried about? Would it solve the issue if we allowed operations that were enabled in the first place to be selectively enabled, but operations that were originally disabled cannot be enabled?

Reply:
I am worried about a situation where a vault uses some sort of a special hook, i.e. KYC-related. If we don't check that the previous hook target was zero or operation wasn't hooked previously, the guardian may enable the operation that previously could only be invoked according to some special rules, i.e. the KYC check.

>Would it solve the issue if we allowed operations that were enabled in the first place to be selectively enabled, but operations that were originally disabled cannot be enabled?

I think it would. It just seemed simpler to check the hook target, but I can implement this.